### PR TITLE
Implemented participant-listing for issue-pages (Fixes #2377)

### LINKED ISSUE #2377: Add participants to issue tracker
In the issue tracker in github, when you have an issue open, there is a list of "contributers" to the discussion on the right hand side.

Is it possible to add this to Gogs?

### PR DESCRIPTION
**Pull request by bkcsoft**
This implements #2377 

gogs_participants

### CODE REVIEW

**Comment by alimy** [comment]
@bkcsoft nice job

**Comment by alimy** [comment]
@bkcsoft but when i build gogs from your source then create a issue in a repository it raise `template: repo/issue/view_content:320:46: executing "repo/issue/view_content" at <len .Participants>: error calling len: len of untyped nil`. the issue is create success but raise error of `template: repo/issue/view_content:320:46: executing "repo/issue/view_content" at <len .Participants>: error calling len: len of untyped nil` too when open it.
screenshot from 2016-01-20 19-49-00
screenshot from 2016-01-20 19-49-05
screenshot from 2016-01-20 19-49-13

**Comment by bkcsoft** [comment]
Are you sure you have a clean build? because I can't reproduce that :/

**Comment by mhartkorn** [comment]
I cannot reproduce the template error, either.

A freshly created issue is missing the creator as a participant until they comment.

**Comment by alimy** [comment]
I build from @bkcsoft  's source code to test this feature use command below: 
```
>$ go get  -u -v -tags "sqlite pam cert" github.com/bkcsoft/gogs
>$ cd $GOPATH/src/github.com/bkcsoft/gogs
>$ git checkout feature/participants 
>$ go build -v -tags "sqlite pam cert"
>$ ./gogs web
```
it is a new install of gogs. I login gogs and create a repository then new a issue for test this feature,a error raise `template: repo/issue/view_content:320:33: executing "repo/issue/view_content" at <len .Participants>: error calling len: len of untyped nil` for me that i comment above.

Is it a problem or anything other?

**Comment by bkcsoft** [comment]
@alimy Try this before `go build`:
```
find . -name "*.go" -type f -exec sed -i 's;github.com/gogits/gogs;github.com/bkcsoft/gogs;g' {} \;
```

**Comment by mhartkorn** [comment]
@alimy: A bit off-topic maybe but you can just use git to checkout pull requests. You don't need to clone the whole repo of @bkcsoft. In this case for PR #2444 you can do this:

    $ cd $GOPATH/src/github.com/gogits/gogs
    $ git fetch origin pull/2444/head:pr-2444
    $ git checkout pr-2444

**Comment by bkcsoft** [comment]
@mhartkorn Fixed the loop-issue.
WRT the OP not showing up before he/she comments, that is unintentionally intentional :laughing: IMO OP shouldn't be listed until he/she has commented and didn't think of it as an error, but now that I compare to how Github looks I see that it is wrong. I'll fix that in a bit.

Also TIL on pulling PRs directly :smile:

**Comment by alimy** [comment]
@bkcsoft  I  `go build `  after    

```
find . -name "*.go" -type f -exec sed -i 's;github.com/gogits/gogs;github.com/bkcsoft/gogs;g' {} \;
```   

yes, now work well.:smile: :thumbsup:

**Comment by alimy** [comment]
@mhartkorn 
> @alimy: A bit off-topic maybe but you can just use git to checkout pull requests. You don't need to clone the whole repo of @bkcsoft. In this case for PR #2444 you can do this:...

this is cool :thumbsup: :smile:

**Comment by bkcsoft** [comment]
@exmex Correct, and fixed :)

**Comment by bkcsoft** [comment]
and rebased on `develop`

File: routers/repo/issue.go [hunk 1]
```diff
@@ -617,9 +619,21 @@ func ViewIssue(ctx *middleware.Context) {
 			}
 
 			marked[comment.PosterID] = comment.ShowTag
+
+			already_added := false
```
Comment (Unknwon) [inline routers/repo/issue.go]: Maybe renamed to `hasIncluded`?

Comment (bkcsoft) [inline routers/repo/issue.go]: sorry, old habit from C :wink:

Comment (bkcsoft) [inline routers/repo/issue.go]: is `isAdded` okay? since they're "added" to a list, not "included" :smiley:

**Comment by Unknwon** [comment]
Merging.

**Pull request merged**